spec.md

```diff
@@ -1663,6 +1708,398 @@ message GetCapacityResponse {
+  // If specified, a volume group will be created from a list of
+  // existing volumes.
+  // This field is OPTIONAL.
+  // repeated string volume_id = 5;
+}
+
+message CreateVolumeGroupResponse {
+  option (alpha_message) = true;
+
+  // Contains all attributes of the newly created volume group.
+  // This field is REQUIRED.
+  VolumeGroup volume_group = 1;
+}
+
+message VolumeGroup {
+  option (alpha_message) = true;
+
+  // The identifier for this volume group, generated by the plugin.
+  // This field is REQUIRED.
+  string volume_group_id = 1;
+
+  // Opaque static properties of the volume group.
+  // This field is OPTIONAL.
+  map<string, string> volume_group_context = 2;
+
+  // Underlying volumes in this group. The same definition in CSI
+  // Volume.
+  // This field is REQUIRED.
+  // To support the creation of an empty group, this list can be empty.
+  // However, this field is not empty in the following cases:
+  // - Response from ListVolumeGroups or GetVolumeGroup if the
+  //   VolumeGroup is not empty.
+  // - Response from ModifyVolumeGroup if the VolumeGroup is not
+  //   empty after modification.
+  // - Phase 2: Create a new volume group from a source group snapshot.
+  // - Phase 2: Create a new volume group and add a list of existing
+  //   volumes to the group.
+  repeated .csi.v1.Volume volumes = 3;
+}
+```
+
+#### `CreateVolume`
+
+TODO
+
+#### `DeleteVolumeGroup`
+
+**ALPHA FEATURE**
+
+```protobuf
+message DeleteVolumeGroupRequest {
+  option (alpha_message) = true;
+
+  // The ID of the volume group to be deprovisioned.
+  // This field is REQUIRED.
+  string volume_group_id = 1;
+
+  // Secrets required by plugin to complete volume group deletion
+  // request.
+  // This field is OPTIONAL. Refer to the `Secrets Requirements`
+  // section on how to use this field.
+  map<string, string> secrets = 2 [(csi_secret) = true];
+}
+
+message DeleteVolumeGroupResponse {
+  option (alpha_message) = true;
+  // Intentionally empty.
+}
+```
+
+#### `ModifyVolumeGroup`
+
+**ALPHA FEATURE**
+
+This RPC will be called by the CO to modify an existing volumegroup on behalf of a user. volume_ids provided in the ModifyVolumeGroupRequest will be compared to the ones in the existing VolumeGroup. New volume_ids in the modified VolumeGroup will be added to the VolumeGroup. Existing volume_ids not in the modified VolumeGroup will be removed from the VolumeGroup. If volume_ids is empty, the VolumeGroup will be removed of all existing volumes. This operation MUST be idempotent.
+
+To support ModifyVolumeGroup, the Kubernetes VolumeGroup controller will be implemented to have a desired state of the world and an actual state of the world. The desired state of the world contains VolumeGroups with the desired PVCList while the actual state of the world contains VolumeGroups with the actual PVCList. The controller will try to reconcile the two by handling adding and removing multiple PVCs through a single CSI RPC call each time.
```

Comment:
This is very Kubernetes specific, I don't think it is suitable to mention in the CSI spec. Maybe phrase it like

> To support ModifyVolumeGroup, a CO implement to have a desired state of the world and an actual state of the world.

_PVCList_ should probably be more like _list of volume-IDs_